### PR TITLE
Added dark-mode feature

### LINKED ISSUE #1: Add Dark Mode Feature to the Virtual Bubble Wrap
### Description:

We would like to implement a **Dark Mode** option for the Virtual Bubble Wrap website, which respects the user's system-wide dark mode setting using the `prefers-color-scheme` media query.

#### **Goals**:
- Automatically detect the user's system dark mode preference using `prefers-color-scheme`.
- Add a toggle button or switch that allows users to manually override the system preference.

#### **Suggested Steps**:
1. Use the `prefers-color-scheme` CSS media query to apply dark mode based on the user's system setting.
2. Create a toggle button or switch to allow users to manually switch between dark and light mode.

### PR DESCRIPTION
**Pull request by SuvineetSingh**
### Description
This PR adds a dark mode toggle component to the app. The feature includes:
- A new toggle switch to switch between light and dark themes.
- Automatic theme detection using the `prefers-color-scheme` media query.
- CSS updates for both light and dark themes.

### Related Issues
Fixes #1 

### Changes Made
- Added a `DarkMode.js` component with logic to handle theme changes.
- Updated `DarkMode.css` to define styles for both themes.
- Modified `App.js` to include the DarkMode component.

### CODE REVIEW

File: src/App.css
Comment (sirisayshello) [inline src/App.css:1]: I can't see that the changes in this file are used or applied in the code, so they could therefore be discarded.

File: src/DarkMode.test.js
Comment (sirisayshello) [inline src/DarkMode.test.js:1]: Super nice that you made some tests!

File: src/components/DarkMode.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      setDark();
+    } else {
+      setLight();
+    }
+
+    // listen for changes in system preferences and update theme if no localStorage value
+    const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");
+    const handleChange = (e) => {
+      if (!localStorage.getItem("theme")) {
+        // mediaQuery.matches ? setDark() : setLight();
```
Comment (sirisayshello) [inline src/components/DarkMode.js]: Since this is commented out, you could remove this line.

**Review by sirisayshello — CHANGES_REQUESTED** [review]
Awesome that you wanted to contribute!

Just had a few notes, but if you sort that out we can get this merged. ✌️ 

Great job!

File: src/App.css
Comment (SuvineetSingh) [reply src/App.css:1]: noted and I just want to say thank you for all the criticism. 👌

File: src/components/DarkMode.js (hunk 1 above)
Comment (SuvineetSingh) [reply src/components/DarkMode.js]: I removed this, and updated the code.

File: src/DarkMode.test.js
Comment (SuvineetSingh) [reply src/DarkMode.test.js:1]: thanks!

**Comment by SuvineetSingh** [comment]
Let me know If I can make any further changes!

**Review by sirisayshello — APPROVED** [review]

**Comment by sirisayshello** [comment]
@SuvineetSingh Looks good, let's merge!

**Comment by SuvineetSingh** [comment]
Thanks @sirisayshello for giving me this opportunity. I wanted to ask you if you have any comments, criticism about my work?

**Pull request merged**